src/components/SamlConfiguration/index.jsx

```diff
@@ -1,279 +1,57 @@
-    transformedData.append('slug', this.props.enterpriseSlug);
-    transformedData.append('enterprise_customer_uuid', this.props.enterpriseId);
-    try {
-      const response = await LmsApiService.postNewProviderConfig(transformedData);
-      this.setState({ providerConfig: response.data });
-      return undefined;
-    } catch (error) {
-      return this.handleErrors(error);
-    }
-  }
-
-  /**
-   * Updates existing provider configuration, then updates this list with the response.
-   * Returns if there is an error.
-   * @param {FormData} formData
-   * @param {string} pid // provider ID generated and stored in database.
-   */
-  updateProviderConfig = async (formData, pid) => {
-    const transformedData = snakeCaseFormData(formData);
-    transformedData.append('name', this.props.enterpriseName);
-    transformedData.append('slug', this.props.enterpriseSlug);
-    transformedData.append('enterprise_customer_uuid', this.props.enterpriseId);
-    try {
-      const response = await LmsApiService.updateProviderConfig(transformedData, pid);
-      this.setState({ providerConfig: response.data });
-      return undefined;
-    } catch (error) {
-      return this.handleErrors(error);
-    }
-  }
-
-  /**
-   * Permanently deletes a provider configuration, then updates this list with the response.
-   * Returns if there is an error.
-   * @param {string} pid // provider ID generated and stored in database.
-   */
-  deleteProviderConfig = async (pid) => {
-    try {
-      await LmsApiService.deleteProviderConfig(pid, this.props.enterpriseId);
-      this.setState({ providerConfig: undefined });
-      return undefined;
-    } catch (error) {
-      return this.handleErrors(error);
-    }
-  }
-
-  /**
-   * Creates new third party provider data entry in db and refreshes list with the result.
-   * Returns if there is an error.
-   * @param {FormData} formData
-   */
-  createProviderData = async (formData) => {
-    formData.append('fetchedAt', moment().format('YYYY-MM-DDThh:mm:ss'));
-    formData.append('enterpriseCustomerUuid', this.props.enterpriseId);
-    const transformedData = snakeCaseFormData(formData);
-    try {
-      const response = await LmsApiService.createProviderData(transformedData);
-      this.setState({ providerData: response.data });
-      return undefined;
-    } catch (error) {
-      return this.handleErrors(error);
-    }
-  }
-
-  /**
-   * Deletes a provider data entry in db and refreshes list with the result.
-   * Returns if there is an error.
-   * @param {string} pdid // provider data id generated & stored in database.
-   */
-  deleteProviderData = async (pdid) => {
-    try {
-      await LmsApiService.deleteProviderData(pdid, this.props.enterpriseId);
-      this.setState({ providerData: undefined });
-      return undefined;
-    } catch (error) {
-      return this.handleErrors(error);
-    }
+    options.push({label: '-- choose a configuration --', value: '', hidden: true})
```

Comment:
TODO: pull existing config in as a prop from ProviderConfig(parent) component and use that as the default here instead of this (unless their existing config doesn't get converted to "public" I guess)